src/helpers/utils.ts

```diff
@@ -149,3 +151,27 @@ export const paramsReducer = (
       return removeParams(existingParams, newParams);
   }
 };
+
+/**
+ * Stringifies value to '' if the value is null
+ */
+export const replaceNullWithEmptyString = (_: any, value: any) =>
+  value === null ? '' : value;
+
+/*
+ * Stringifies all JSON field values in the specified fields array.
+ */
+export const stringifyJsonFields = (
+  object: Record<string, any>,
+  fields: string[] = [],
+): Record<string, any> => {
+  const modifiedObject = { ...object };
+
+  for (const field of fields) {
+    const value = _.get(object, field);
+    const newValue = JSON.stringify(value, replaceNullWithEmptyString);
+    _.set(modifiedObject, field, newValue);
+  }
+
+  return modifiedObject;
+};
```

Comment:
suggestion (non-blocking): since modifiedObject is being used as an accumulator, we could simplify this with a reduce:

```suggestion
export const stringifyJsonFields = (
  object: Record<string, any>,
  fields: string[] = []
): Record<string, any> => {
  return fields.reduce((acc, field) => {
    const value = _.get(object, field);
    const newValue = JSON.stringify(value, replaceNullWithEmptyString);
    return _.set({ ...acc }, field, newValue);
  }, object);
};
```

Reply:
question:

Wouldn't that modify my input `object`? I'm being careful to note mutate the original object with this function.

Reply:
hmmm yeah since it's a reference, you might be messing with the OG, good point